tests/testthat/test-derive_var_atoxgr.R

```diff
@@ -3191,3 +3191,2836 @@ test_that("derive_var_atoxgr, test 81: CTCAEv4 Hypophosphatemia", {
+  bind_rows(
+    expected_ancd_daids_ge2d,
+    expected_ancd_daids_le1d
+  )
+
+# Set lab date to missing fo each type, ie SEX is M, F or missing
+expected_ancd_daids_noage <- expected_ancd_daids %>%
+  filter(TESTNUM %in% c(20, 31)) %>%
+  mutate(
+    LBDT = if_else(TESTNUM == 20, NA, LBDT),
+    BRTHDT = if_else(TESTNUM == 31, NA, BRTHDT),
+    ATOXGRL = NA,
+    TESTNUM = case_when(
+      TESTNUM == 20 ~ 34,
+      TESTNUM == 31 ~ 35
+    )
+  )
+
+expected_ancd_daids <- expected_ancd_daids %>%
+  bind_rows(expected_ancd_daids_noage)
+
+
+input_ancd_daids <- expected_ancd_daids %>%
+  select(-ATOXGRL)
+
+
+## Test 115: DAIDS ANC Low ----
+test_that("derive_var_atoxgr Test 115: DAIDS ANC Low", {
+  actual_ancd_daids <- derive_var_atoxgr_dir(
+    input_ancd_daids,
+    new_var = ATOXGRL,
+    meta_criteria = atoxgr_criteria_daids,
+    tox_description_var = ATOXDSCL,
+    criteria_direction = "L",
+    get_unit_expr = AVALU
+  )
+
+  expect_dfs_equal(
+    base = expected_ancd_daids,
+    compare = actual_ancd_daids,
+    keys = c("TESTNUM")
+  )
+})
+
+
+### Fibrinogen Decreased
+### Grade 4: <0.5 g/L OR < 0.25 x LLN
+### Grade 3: 0.5 to <0.75 g/L OR 0.25 to < 0.50 x LLN
+### Grade 2: 0.75 to <1 g/L OR ≥ 0.50 to < 0.75 x LLN
+### Grade 1: 1 to < 2 g/L OR 0.75 to < 1.00 x LLN
+
+## Test 116: DAIDS Fibrinogen Decreased ----
+test_that("derive_var_atoxgr Test 116: DAIDS Fibrinogen Decreased", {
+  expected_fibd_daids <- tibble::tribble(
+    ~ATOXDSCL,              ~AVAL, ~ANRLO, ~AVALU, ~ATOXGRL, ~TESTNUM,
+    "Not a term",           2,     1,      "g/L",  NA,       1,
+    NA_character_,          2,     1,      "g/L",  NA,       2,
+    "Fibrinogen Decreased", 2,     1,      "g/dL", NA,       3,
+    # test first half of criteria
+    "Fibrinogen Decreased", 0.49,  1,      "g/L",  "4",      4,
+    "Fibrinogen Decreased", 0.5,   1,      "g/L",  "3",      5,
+    "Fibrinogen Decreased", 0.74,  1,      "g/L",  "3",      6,
+    "Fibrinogen Decreased", 0.75,  1,      "g/L",  "2",      7,
+    "Fibrinogen Decreased", 0.99,  1,      "g/L",  "2",      8,
+    "Fibrinogen Decreased", 1,     1,      "g/L",  "1",      9,
+    "Fibrinogen Decreased", 1.99,  1,      "g/L",  "1",      10,
+    "Fibrinogen Decreased", 2,     1,      "g/L",  "0",      11,
+    # test second half of criteria
+    "Fibrinogen Decreased", 0.74,  3,      "g/L",  "4",      12,
+    "Fibrinogen Decreased", 0.75,  3,      "g/L",  "3",      13,
+    "Fibrinogen Decreased", 1.49,  3,      "g/L",  "3",      14,
+    "Fibrinogen Decreased", 1.5,   3,      "g/L",  "2",      15,
+    "Fibrinogen Decreased", 2.24,  3,      "g/L",  "2",      16,
+    "Fibrinogen Decreased", 2.25,  3,      "g/L",  "1",      17,
+    "Fibrinogen Decreased", 2.99,  3,      "g/L",  "1",      18,
+    "Fibrinogen Decreased", 3,     3,      "g/L",  "0",      19,
+    # TEST for missing values
+    "Fibrinogen Decreased", 0.49,  NA,     "g/L",  "4",      20,
```

Comment:
```suggestion
    "Fibrinogen Decreased", 0.49,  NA,     "g/L",  NA,      20,
```

Wonder how this is getting through

Reply:
This one is OK, as AVAL < 0.5 g/L is satisfies grade 4. (We don't need to look at ANRLO).